go/mobile/identity.go

```diff
@@ -131,15 +130,23 @@ func NewIdentityLoad(storePath, pass, web3Url string, checkTicketsPeriodMilis in
 	// Init Identity
 	iden := &Identity{
 		id:          holdr,
-		eventSender: e,
 		storage:     storage,
 		tickets:     NewTickets(storage.WithPrefix([]byte(ticketPrefix))),
+		stopTickets: make(chan bool),
+		eventQueue:  make(chan Event, 10),
 	}
-	iden.stopTickets = make(chan bool)
-	go iden.tickets.CheckPending(iden, iden.eventSender, time.Duration(checkTicketsPeriodMilis)*time.Millisecond, iden.stopTickets)
+	go iden.tickets.CheckPending(iden, iden.eventQueue, time.Duration(checkTicketsPeriodMilis)*time.Millisecond, iden.stopTickets)
 	return iden, nil
 }
 
+// GetNextEvent returns the oldest event that has been generated.
+// Note that each event can only be retireved once.
+// Note that this function is blocking and potentially for a very long time.
+func (i *Identity) GetNextEvent() *Event {
```

Comment:
I think it would be interesting that the receiver of the events is able to mark them as "read".  And while they are not marked as "read", they keep appearing in `GetNextEvent`.  This allows the app to be closed when there's something in the queue without losing the events in the queue.

Reply:
We can discuss this and if we agree on it, do it in a future PR.

Reply:
TODO: #35